### DISCUSSION TITLE
New feature request - Button to exec external script/application

### DISCUSSION THREAD

**Discussion by xes** [gdyuldin/x6100_gui#72] [Ideas]
A request for the future and for all linux lovers..

To expand the application functions and exec additional applications or scripts, it would be fantastic to be able to define and add a customizable button to the application to exec an external command. 

An option in the settings like AddButton:  Button Name / Script to run. 

I can imagine a bluetooth pairing script, start a vnc server, open a terminal, a Zenity/Yad/Xdialog.... unlimited possibilities easily accessible!

**Comment by gdyuldin** [comment]
I'm not sure. Bluetooth pairing should be a part of the application, if we want to use external devices with X6100. VNC server is useless - there is no X, app draw direct to the frame buffer. Terminal useful, if you have at least a keyboard. Also, device screen relatively small. What will happens with GUI application, when terminal is opened?
Anyway - you can connect X6100 using wifi, ssh to it and launch any command, if you need.

**Reply by ericb2B** [reply]
There are more application for custom button, i think it s a great idear
